### PR TITLE
[chore] 0.4.2 release

### PR DESCRIPTION
**Pull request by anupambhatnagar**
## What does this PR do?
Publishing release v0.4.2.

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Review by tmarkstrum — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## NEXT - TBD
+## [0.4.2] - 2021-11-08
```
Comment (anj-s) [inline CHANGELOG.md:7]: There were more changes that happened this month. I have a PR that I need to merge once VPN works for me. That should update the CHANGELOG.

File: README.md [hunk 2]
```diff
@@ -22,8 +22,7 @@ FairScale was designed with the following values in mind:
 
 ## What's New:
 
-* September 2021 [`master` branch renamed to `main`](https://github.com/github/renaming).
-* September 2021 [fairscale 0.4.1 was released](https://github.com/facebookresearch/fairscale/releases/tag/v0.4.1).
+* November 2021 [fairscale 0.4.2 was released](https://github.com/facebookresearch/fairscale/releases/tag/v0.4.2).
```
Comment (anj-s) [inline README.md:25]: Can we maintain the last 5 news items? Having a single one looks odd.

**Review by min-xu-ai — APPROVED** [review]
LGTM! Thanks and agree with @anj-s's comments too.

File: README.md (hunk 2 above)
Comment (anupambhatnagar) [reply README.md:25]: @anj-s I'll retain the Sep 2021 items. please provide two more items.

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # FairScale should only depends on torch, not things higher level than torch.
-torch >= 1.7.0
+torch >= 1.8.0
```
Comment (anj-s) [inline requirements.txt:2]: should we revert this change and make it with the CI config PR?

Comment (anupambhatnagar) [reply requirements.txt:2]: yes, I think that would be appropriate.

File: README.md (hunk 2 above)
Comment (anj-s) [reply README.md:25]: 1. We have a new experimental API called the [LayerwiseMemoryTracker](https://fairscale.readthedocs.io/en/latest/tutorials/layer_memory_tracking.html) to help track, visualize and suggest fix for memory issues occurring during the forward/backward pass of your models. 
2. Introducing [SlowMoDistributedDataParallel](https://fairscale.readthedocs.io/en/latest/tutorials/slowmo_ddp.html) API, a distributed training wrapper that is useful on clusters with slow network interconnects (eg Ethernet).

File: CHANGELOG.md (hunk 1 above)
Comment (anupambhatnagar) [reply CHANGELOG.md:7]: done

File: README.md (hunk 2 above)
Comment (anupambhatnagar) [reply README.md:25]: thanks, added to the readme.

**Pull request merged**